### PR TITLE
add articles with pages and individual article pages

### PR DESCRIPTION
**Pull request by shamelhok**
click on all articles to view articles page

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    if(!order) order = ''
+    if(!limit) limit = ''
+    if(!p) p = ''
+    return fetch('https://nc-news-shamel.herokuapp.com/api/articles?sort_by='
+    +sort_by+'&order='+order+'&limit='+limit+'&p='+p+topic).then(res=>{
+        return res.json()
+    })
+}
+
+function fetchArticleById(article_id){
```
Comment (dappernerddesigns) [inline src/api.js:15]: Why are there api calls for functionality that is not on the ticket?

File: src/api.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+function fetchArticles(obj={}){
+    let {topic,sort_by,order,limit,p}= obj
+    if(!topic){ topic = ''
```
Comment (dappernerddesigns) [inline src/api.js:3]: Complex sorts on the api are not needed until ticket 10

File: src/components/Article.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        {        })
+    const {article_id}=useParams()
+    useEffect(()=>{
+        fetchArticleById(article_id).then(({article})=>{
+            setCurrentArticle(article)
+        })
+    },[currentArticle])
+    return (<div>
+        article :
+        <ArticleTitleCard {...currentArticle}/>
```
Comment (dappernerddesigns) [inline src/components/Article.js:17]: why is spread needed here? Viewing an object should in no way mutate it

File: src/components/Article.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { useParams } from "react-router-dom";
+import { fetchArticleById } from "../api";
+import ArticleTitleCard from "./ArticleTitleCard";
+
+export default function Article(){
+    const[currentArticle,setCurrentArticle]= useState(
+        {        })
+    const {article_id}=useParams()
+    useEffect(()=>{
+        fetchArticleById(article_id).then(({article})=>{
```
Comment (dappernerddesigns) [inline src/components/Article.js:11]: Why is there a component for fetching by ID on ticket 4?

File: src/index.js [hunk 5]
```diff
@@ -3,11 +3,14 @@ import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import { BrowserRouter } from 'react-router-dom';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
   <React.StrictMode>
+    <BrowserRouter>
```
Comment (dappernerddesigns) [inline src/index.js:11]: Where are your routes?

**Review by dappernerddesigns — APPROVED** [review]
I am approving this PR purely because the functionality works.

Please don't add more code than is necessary per ticket, you have submitted functionality for two tickets here under the guise of one.

File: src/components/Article.js (hunk 3 above)
Comment (shamelhok) [reply src/components/Article.js:17]: passing props

**Pull request merged**